main.go

```diff
@@ -74,6 +80,34 @@ func SaveMessageToFile(body []byte, outputDir string, counter uint) {
 	fmt.Println(filePath)
 }
 
+func SavePropertiesToFile(msg amqp.Delivery, outputDir string, counter uint) {
+	var b bytes.Buffer
+
+	WriteStringProperty(&b, "content_type", msg.ContentType)
+	WriteStringProperty(&b, "content_encoding", msg.ContentEncoding)
+	WriteUInt8Property(&b, "delivery_mode", msg.DeliveryMode)
+	WriteUInt8Property(&b, "priority", msg.Priority)
+	WriteStringProperty(&b, "correlation_id", msg.CorrelationId)
+	WriteStringProperty(&b, "reply_to", msg.ReplyTo)
+	WriteStringProperty(&b, "expiration", msg.Expiration)
+	WriteStringProperty(&b, "message_id", msg.MessageId)
+
+	if !msg.Timestamp.IsZero() {
+		WriteStringProperty(&b, "timestamp", msg.Timestamp.String())
+	}
+
+	WriteStringProperty(&b, "type", msg.Type)
+	WriteStringProperty(&b, "user_id", msg.UserId)
+	WriteStringProperty(&b, "app_id", msg.AppId)
+
+	if b.Len() > 0 {
```

Comment:
In case there are no properties you don't create the file - that might be confusing when some messages have properties and some doesn't. I don't see what's wrong with an empty file in such a case.